File: server/dotnet/sdk-dotnet-example/Business/Domain/Payments/PaymentDetails/PaymentStatusOutput.cs
using System.Collections.Generic;

namespace Business.Domain.Payments.PaymentDetails;

public class PaymentStatusOutput
{
    public IList<APIError>? Errors { get; set; }
    
    public bool? IsAuthorized { get; set; }

    public bool? IsCancellable { get; set; }

    public bool? IsRefundable { get; set; }

    public string? StatusCategory { get; set; }

    public int? StatusCode { get; set; }
    
    public string? StatusCodeChangeDateTime { get; set; }
}

File: server/ruby/app/business/domain/payments/payment_details/payment_status_output.rb
module Business
  module Domain
    module Payments
      module PaymentDetails
        class PaymentStatusOutput
          attr_accessor :errors, :is_authorized, :is_cancellable, :is_refundable,
                        :status_category, :status_code, :status_code_change_date_time

          def initialize(errors: nil, is_authorized: nil, is_cancellable: nil, is_refundable: nil,
                         status_category: nil, status_code: nil, status_code_change_date_time: nil)
            @errors = errors
            @is_authorized = is_authorized
            @is_cancellable = is_cancellable
            @is_refundable = is_refundable
            @status_category = status_category
            @status_code = status_code
            @status_code_change_date_time = status_code_change_date_time
          end

          def to_h
            {
              'errors' => @errors&.map(&:to_h),
              'isAuthorized' => @is_authorized,
              'isCancellable' => @is_cancellable,
              'isRefundable' => @is_refundable,
              'statusCategory' => @status_category,
              'statusCode' => @status_code,
              'statusCodeChangeDateTime' => @status_code_change_date_time
            }
          end
        end
      end
    end
  end
end